docs/ExternalResources.md

```diff
@@ -0,0 +1,164 @@
+case we break the tie arbitrarily.
+-->
+
+<!--
+The textual descriptions should be written in a way that makes it clear
+which topics are discussed, and what a potential contributor to Swift
+will gain from it. This is usually different from the talk's abstract.
+-->
+
+## Contributing Guides and Tips
+
+- [Steps for first PR and asking questions][] by Varun Gandhi (Swift forum
+  comment, Dec 2019).
+- [Contributing to Swift for the First Time][] by Robert Pieta (talk, Jun 2019):
+  This talk describes Robert's experience contributing to
+  `swift-corelibs-foundation` and later `swift`, providing a step-by-step guide
+  for new contributors to get their feet wet.
+- [Contributing to Swift compiler][] by Yusuke Kita (lightning talk, Apr 2019).
+- Contributing to Swift ([Part 1][Contributing to Swift - Part 1],
+  [Part 2][Contributing to Swift - Part 2]) by Suyash Srijan (blog post series,
+  Dec 2018).
+- [Setting up a compile-edit-test cycle][] by Robert Widmann (Swift forum
+  comment, May 2018)
+- [Becoming an Effective Contributor to Swift][] by Harlan Haskins and Robert
+  Widmann (talk, Apr 2018): Covers the following topics:
+  - The organization of different projects part of the Swift monorepo.
+  - How to set up an environment for building Swift.
+  - Common utilities and compiler flags for debugging different stages.
+  - Common build configurations.
+  - Tips and tricks for testing: early exiting, filtering tests,
+    testing Swift and Objective-C together, using `PrettyStackTrace`.
+  - A live demo fixing a crash:
+    - Copying program arguments into Xcode for debugging.
+    - Using `dump()` for print debugging.
+- [Getting Started with Swift Compiler Development][] by Brian Gesiak (blog post
+  series, Aug 2017 - Jun 2018)
+- [Contributing to Open Source Swift][] by Jesse Squires (talk, Mar 2016):
+  Covers the following topics:
+  - The overall compiler pipeline.
+  - The organization of different projects part of the Swift monorepo,
+    including some "difficulty levels" for different projects.
+  - Tips for contributing effectively.
+
+[Steps for first PR and asking questions]:
+https://forums.swift.org/t/getting-started-with-swift-compiler-development/31502/2
+[Contributing to Swift for the First Time]: https://youtu.be/51j7TrFNKiA
+[Contributing to Swift compiler]: https://youtu.be/HAXJsgYniqE
+[Contributing to Swift - Part 1]: https://medium.com/kinandcartacreated/contributing-to-swift-part-1-ea19108a2a54
+[Contributing to Swift - Part 2]:
+https://medium.com/kinandcartacreated/contributing-to-swift-part-2-efebcf7b6c93
+[Setting up a compile-edit-test cycle]: https://forums.swift.org/t/need-a-workflow-advice/12536/14
+[Becoming an Effective Contributor to Swift]: https://youtu.be/oGJKsp-pZPk
+[Getting Started with Swift Compiler Development]: https://modocache.io/getting-started-with-swift-development
+[Contributing to Open Source Swift]: https://youtu.be/Ysa2n8ZX-YY
+
+## AST
+
```

Comment:
@varungandhi-apple Would content about libSyntax be useful here? There is this great talk by @harlanhaskins about the basics of the compiler pipeline and libSyntax. Here https://www.youtube.com/watch?v=5ivuYGxW_3M